util/httputil/http_util.go

```diff
@@ -84,3 +86,49 @@ func AuthTransport(auth *config.Auth, transportConfig *http.Transport) (http.Rou
 
 	return newAuthRoundTripper(auth, transportConfig), nil
 }
+
+func GuessKialiURL(r *http.Request) string {
```

Comment:
I wonder if there are corner cases where this would not work. Perhaps as a safety net we could add a field in openid config to override the redirection URL ? GuessKialiURL would be called only if this field isn't present?

Or are you confident that it should work in any case?

Reply:
I was thinking about it yesterday.
The "success" of this function depends 100% on the client. If it doesn't send the HTTP headers, it's impossible to determine the public URL.

So, I think this is not totally safe. Thus, I'll add the new config value. I think I'll put it under the `Server` next to the `WebRoot` config which is closely related.

Reply:
Fixed. Now there are settings to configure all parts of the public URL